### DISCUSSION TITLE
async for ctx.master.commands.call?

### DISCUSSION THREAD

**Discussion by hackerliang** [mitmproxy/mitmproxy#5147] [Q&A]
I just write a function
```python
from mitmproxy import ctx
from mitmproxy import HTTP

def response(flow: http.HTTPFlow) -> None:
    ........
    resend_flow = flow.copy()
    ctx.master.commands.call("replay.client", [resend_flow])
```

But this is too slow. The next request needs to wait for the `ctx.master.commands.call` of the previous request to complete before executing. Any ideas?

**Comment by mhils** [comment]
Take a look at https://github.com/mitmproxy/mitmproxy/blob/main/examples/addons/nonblocking.py!
